scheduler.py

```diff
@@ -71,3 +76,27 @@ def get_task_completion(self, task):
     def sigterm_handler(self, signum, frame):
         logging.info("Exiting gracefully.")
         sys.exit(0)
+
+    def get_worker(self):
+        if self.schedulerMode == constants.SCHEDULINGMODE_RANDOM:
+            logging.info(f"Getting worker for {self.schedulerMode}")
+            if self.globalIncrementalWorkerId > 1:
+                random_worker_id = random.randint(0, self.globalIncrementalWorkerId - 1)
+                with self.workerMapLock:
+                    return self.workers[random_worker_id]
+            else:
```

Comment:
we will hit the else, if there is only one worker. The if condition is self.globalIncrementalWorkerId > 1. Please try running SCHEDULINGMODE_RANDOM with only one worker locally